test/src/test_vaildate_generated_sq.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+import os
+import json
+import jsonschema
+
+schema_file = "json-schema/StructuredQuery-schema-v3.json"
+generated_sqs_dir = "test/resources/generated-sqs/"
+
+@pytest.fixture
+def schema():
+    loaded_schema = None
+    with open(schema_file, 'r') as file:
+        loaded_schema = json.load(file)
+    return loaded_schema
+
+def test_validate_sqs(schema):
+    if not os.path.exists(generated_sqs_dir):
+        return
+    
+    sqs = []
+    for file in os.listdir(generated_sqs_dir):
+        filename = os.fsdecode(file)
+        if not filename.endswith(".json") : 
+            continue
+        
+        with open(f"{generated_sqs_dir}{filename}", 'r') as f:
+            sqs.append(json.load(f))
+    
+    for sq in sqs:
+        jsonschema.validate(sq, schema)
```

Comment:
Does the test fail if the validation fails?

Reply:
yes